### PR TITLE
Stringmanip

### PR DESCRIPTION
**Pull request by haticeaydinn**
1.0.2. String Manipulation

### CODE REVIEW

File: Chapter 1. Programming/Module 0. Programming I/1.0.2. String Manipulation.ipynb [hunk 1]
```diff
@@ -0,0 +1,938 @@
+{
```
Comment (joao-binenbojm) [inline Chapter 1. Programming/Module 0. Programming I/1.0.2. String Manipulation.ipynb:1]: <p>Check feedback for other notebooks on learning objectives format</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AI-Core/Foundation-Course/pull/30/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (joao-binenbojm) [inline Chapter 1. Programming/Module 0. Programming I/1.0.2. String Manipulation.ipynb:1]: <p>Check feedback for other notebooks on headers</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AI-Core/Foundation-Course/pull/30/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (joao-binenbojm) [inline Chapter 1. Programming/Module 0. Programming I/1.0.2. String Manipulation.ipynb:1]: <p>I would make it more explicit somewhere around here about how negative indices are basically indexing from the end of the array to the beginning. I would also make the differences between :: and : more explicit.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AI-Core/Foundation-Course/pull/30/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (joao-binenbojm) [inline Chapter 1. Programming/Module 0. Programming I/1.0.2. String Manipulation.ipynb:1]: <p>I love the cake! Great idea!!!!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AI-Core/Foundation-Course/pull/30/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (joao-binenbojm) [inline Chapter 1. Programming/Module 0. Programming I/1.0.2. String Manipulation.ipynb:1]: <ul><li>Might be nice to just have a diagram here to just give them a brief overview of how a function is something that takes in an input and gives an output (the simpler the better)</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AI-Core/Foundation-Course/pull/30/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (joao-binenbojm) [inline Chapter 1. Programming/Module 0. Programming I/1.0.2. String Manipulation.ipynb:1]: <p>As I mentioned before, perhaps this should be moved to just before the slicing section</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AI-Core/Foundation-Course/pull/30/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (joao-binenbojm) [inline Chapter 1. Programming/Module 0. Programming I/1.0.2. String Manipulation.ipynb:1]: <p>I would give a bit more context on this one just cause it's a tricky one the first time you came around it, especially for default separators. Maybe one of your infamous diagrams might help!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AI-Core/Foundation-Course/pull/30/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Comment (joao-binenbojm) [inline Chapter 1. Programming/Module 0. Programming I/1.0.2. String Manipulation.ipynb:1]: <p>I would try to explicitly mention what an iterable is in python</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AI-Core/Foundation-Course/pull/30/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (joao-binenbojm) [inline Chapter 1. Programming/Module 0. Programming I/1.0.2. String Manipulation.ipynb:1]: <p>Awesome example and great explanations so far! I should've probably mentioned before but for our notebooks, we generally like to use ## over # to denote a coding challenge so the students know where to type. Very minor thing but I thought you should know</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AI-Core/Foundation-Course/pull/30/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='44'/>

Comment (joao-binenbojm) [inline Chapter 1. Programming/Module 0. Programming I/1.0.2. String Manipulation.ipynb:1]: <p>While this is a good practice exercise, I feel like it should be moved as an extra for branching and iteration. it doesn't have much to do with strings and it's a perfect opportunity to come up with something to make them practice using split, upper, lower, join, index, etc</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AI-Core/Foundation-Course/pull/30/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='49'/>

Comment (haticeaydinn) [reply Chapter 1. Programming/Module 0. Programming I/1.0.2. String Manipulation.ipynb:1]: <p>Actually i added lots of examples for the slicing part :: and : . You can see them end of the section </p><p> </p><p>print(s[3:6])&nbsp;&nbsp;&nbsp;# evaluates to "def", same as s[3:6:1]</p><p>print(s[3:6:2])&nbsp;&nbsp;# evaluates to "df"</p><p>print(s[::])&nbsp;&nbsp;&nbsp;# evaluates to "abcdefgh", same as s[0:len(s):1]</p><p>print(s[:])&nbsp;&nbsp;&nbsp;&nbsp;# evaluates to "abcdefgh", same as s[0:len(s)] step=1 by default</p><p>print(s[::-1])&nbsp;&nbsp;# evaluates to "hgfedbca", same as s[-1:-(len(s)+1):-1]</p><p>print(s[4:1:-2])&nbsp;# evaluates to "ec"</p>

**Pull request merged**